resourcemgr/resourcemgr.c

```diff
@@ -245,7 +246,7 @@ typedef struct {
                                     //  flushed.
     UINT64 connectionId;            // Used to identify which connection owns the object,
                                     // sequence, or session.
-    void *nextEntry; // Next entry in the list; 0 to terminate list.                                
+    RESOURCE_MANAGER_ENTRY_PTR nextEntry; // Next entry in the list; 0 to terminate list.                             
 } RESOURCE_MANAGER_ENTRY;
```

Comment:
RESOURCE_MANAGER_ENTRY is a data structure that keeps all the state data about an object, sequence, or session that the RM needs to track.

The RM maintains and uses a linked list of these structures.

Will Arthur
Intel Corporation
Server Security Firmware
803-216-2101

From: Philip Tricca [mailto:notifications@github.com]
Sent: Thursday, March 24, 2016 2:31 PM
To: 01org/TPM2.0-TSS <TPM2.0-TSS@noreply.github.com>
Cc: Arthur, Will C <will.c.arthur@intel.com>
Subject: Re: [TPM2.0-TSS] Debugability enhancements to RPSOURCE_MANAGER_ENTRY structure (#155)


In resourcemgr/resourcemgr.c<https://github.com/01org/TPM2.0-TSS/pull/155#discussion_r57365462>:

> @@ -245,7 +246,7 @@ typedef struct {

>                                      //  flushed.

>      UINT64 connectionId;            // Used to identify which connection owns the object,

>                                      // sequence, or session.

> -    void *nextEntry; // Next entry in the list; 0 to terminate list.

> +    RESOURCE_MANAGER_ENTRY_PTR nextEntry; // Next entry in the list; 0 to terminate list.

>  } RESOURCE_MANAGER_ENTRY;

From looking just at the structure and it's name it's not entirely clear what a RESOURCE_MANAGER_ENTRY is (to me at least). The structure holds a pile of data and it all seems related to various handles that the RM may be switching in / out so an 'entry' would be a handle then?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/01org/TPM2.0-TSS/pull/155/files/af4ac0b43489264634d724d22170f8646611cf11#r57365462>